sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -55,4 +56,49 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
     } as InstitutionLocation;
     return this.repo.save(programOffering);
   }
+
+  /**
+   * This is to fetch all the Education Offering
+   * that are associated with the Location and Program
+   * @param locationId
+   * @param programId
+   * @returns
+   */
+  async getAllEducationProgramOffering(
+    locationId: number,
+    programId: number,
+  ): Promise<EducationProgramOfferingModel[]> {
+    const educationProgramOfferingResult = await this.repo
+      .createQueryBuilder("education_programs_offerings")
+      .select([
+        "education_programs_offerings.id",
+        "education_programs_offerings.name",
+        "education_programs_offerings.studyStartDate",
+        "education_programs_offerings.studyEndDate",
+        "education_programs_offerings.offeringDelivered",
+      ])
+      .leftJoin(
+        "education_programs_offerings.educationProgram",
+        "educationProgram",
+      )
+      .leftJoin(
+        "education_programs_offerings.institutionLocation",
+        "institutionLocation",
+      )
+      .where(
+        "educationProgram.id = :programId and institutionLocation.id = :locationId",
+        { programId: programId, locationId: locationId },
+      )
+      .getMany();
+
+    return educationProgramOfferingResult.map((educationProgramOffering) => {
+      const item = new EducationProgramOfferingModel();
+      item.id = educationProgramOffering.id;
+      item.name = educationProgramOffering.name;
+      item.studyStartDate = educationProgramOffering.studyStartDate;
```

Comment:
I believe that I faced some issues with the property case when returning an "any" from TypeOrm.
I think that the casing would be as below.
```
educationProgramOffering.studystartdate
```

Reply:
agreed